src/sync/integrations/github.ts

```
@@ -0,0 +1,500 @@
+			const initialMessage = Integration.translateMessage(issueData);
+			// This URL is slightly fabricated, but allows us to reference the
+			// initial message distinct from its issue
+			initialMessage.url = `${initialMessage.url}#`;
+			const sortedComments = _.sortBy(details.comments.data, (comment) => {
+				return moment(comment.created_at).unix();
+			});
+			const tailMessages = _.map(sortedComments, Integration.translateMessage);
+			const extraTags = [params.repo];
+			const thread = Integration.translateThread(issueData, { extraTags });
+			return {
+				messages: _.concat([initialMessage], tailMessages),
+				thread,
+			};
+		}, explodedOptions.retry.retry);
+	}
+
+	public async push(
+		payload: ChatContext,
+		options?: Partial<GitHubOptions>
+	): Promise<ChatContext> {
+		const explodedOptions = this.explodeOptions(options);
+		return await retry(async () => {
+			const thread = payload.thread;
+			const firstMessage = _.first(payload.messages);
+			const noRetry = explodedOptions.github.noRetry;
+			const writtenThread = payload.thread.url
+				? payload
+				: await this.pushThread(thread, firstMessage, noRetry);
+			const replies = payload.thread.url
+				? payload.messages
+				: _.tail(payload.messages);
+			const writtenReplyContexts = await Bluebird.map(
+				replies,
+				async (eachMessage) => {
+					return this.pushMessage(
+						writtenThread.thread,
+						eachMessage,
+						explodedOptions.github.noRetry
+					);
+				},
+				{ concurrency: 1 }
+			);
+			// We flatten this because the response from pushMessage is an array of the messages pushed
+			const writtenReplies = _.flatten(_.map(writtenReplyContexts, 'messages'));
+			return {
+				thread: writtenThread.thread,
+				messages: _.concat(writtenThread.messages, writtenReplies),
+			};
+		}, explodedOptions.retry.retry);
+	}
+
+	public translate(
+		{ body }: AnyResponse['data'],
+		_options?: Partial<GitHubOptions>
+	): Promise<ChatContext[]> {
+		try {
```

This is basically adding a `try` clause over the whole function body, which is the same as not doing it (ie if any error happens on any of these lines, the whole function will reject if you add the `async` keyword).

That being said, I do think we need to add a `try`/`catch` just around the `JSON.parse()` line.

---

We clearly need to touch base on this one.  What would the try/catch around the JSON.parse do?
1) Swallow the error, and mean that some sorts of duff input do error and some sorts don't?
2) Put a meaningful error message on the error?